src/sizeMapping.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @module sizeMapping
+ */
+import * as utils from './utils';
+
+exports.mapSizes = function(adUnit) {
+  if(!isSizeMappingValid(adUnit.sizeMapping)){
+    return adUnit.sizes;
+  }
+  const width = this.getScreenWidth();
+  if(!width) {
+    //size not detected - get largest value set for desktop
+    const mapping = adUnit.sizeMapping.reduce((prev, curr) => {
+      return prev.minWidth < curr.minWidth ? curr : prev;
+    });
+    if(mapping.sizes) {
+      return mapping.sizes;
+    }
+    return adUnit.sizes;
+  }
+  const sizes = adUnit.sizeMapping.find(sizeMapping =>{
+    return width > sizeMapping.minWidth;
+  }).sizes;
+  utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${adUnit.sizes}`);
```

Comment:
If a `sizeMapping` is configured, is that meant to replace the `sizes` property? If so this message will show width as `undefined`